### DISCUSSION TITLE
How to get Workflow url related to the current action?

### DISCUSSION THREAD

**Discussion by ravgeetdhillon** [community/community#27114] [Actions]
<p>I am building an actions workflow. In the action, I would love to get the path of the workflow file that is responsible for running this action?</p>

**Comment by mickeygousset** [comment]
<p>Using the github context object, you can get the display name of the workflow:  ${{ github.workflow }}</p>
<p>But I can’t find anything that gives you the file name.</p>
<p>As far as getting the path, all workflow files are in the .github/workflows folder, so a hack would be to set the display name of your workflow file to be the same as the actual file name, then could get the information you want.</p>

**Comment by ravgeetdhillon** [comment, marked as answer]
<p>Thanks <a class="mention" href="/mickeygousset">@mickeygousset</a>. I have instead used Github API for getting workflows.</p>

Edit: I have posted my answer below. Kindly take a look.

**Reply by Eforen** [reply]
Please say HOW! Don't do the meme! Help others! that's why you have 👎 on this.

**Reply by ravgeetdhillon** [reply]
@Eforen I never got any notifications for this issue that's why I couldn't respond to this issue.

Here's the code that you can use to get the work flow URL:

```js
def get_workflow_url():
    """
    Get Workflow URL responsible for the Action run.
    """
    repo = os.getenv('GITHUB_REPOSITORY')
    github_token = os.getenv('GITHUB_SECRET')
    url = f'https://api.github.com/repos/{repo}/actions/workflows'
    headers = {
        'Accept': 'application/vnd.github.v3+json',
        'Authorization': f'token {github_token}',
    }
    response = requests.get(url, headers=headers)
    if response.status_code != 200:
        return
    workflows = response.json()['workflows']
    for workflow in workflows:
        if workflow['name'] == os.getenv('GITHUB_WORKFLOW'):
            return workflow['html_url']
```

**Reply by abhishekbedi1432** [reply]
Use like this:
```
"${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
```

**Comment by infinisil** [comment]
You can get the workflow ID from the run ID ([reference](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#get-a-workflow-run)):

```bash
workflowId=$(gh api \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  /repos/${{ github.repository }}/actions/runs/${{ github.run_id }} \
  --jq .workflow_id)
```
